tests/pytorch_tests/model_tests/feature_models/reuse_layer_net_test.py

```diff
@@ -12,56 +12,164 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # ==============================================================================
-import torch
+
+from mct_quantizers import PytorchQuantizationWrapper
 from tests.pytorch_tests.model_tests.base_pytorch_test import BasePytorchTest
 
-"""
-This test checks:
-The reuse of a layer in a model.
-"""
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
 
-class ReuseLayerNet(torch.nn.Module):
+class BaseReuseLayerNet(nn.Module):
     def __init__(self):
-        super(ReuseLayerNet, self).__init__()
-        self.conv1 = torch.nn.Conv2d(3, 3, kernel_size=1, stride=1)
-        self.conv2 = torch.nn.Conv2d(3, 3, kernel_size=1, stride=1)
-        self.identity = torch.nn.Identity()
+        super().__init__()
 
-    def forward(self, x, y):
-        x = self.conv1(x)
-        x = self.identity(x)
-        x = self.conv1(x)
-        x = self.identity(x)
+    def forward(self, x):
+        raise NotImplementedError("Subclasses must implement forward method")
+
+class ReuseLayerNet(BaseReuseLayerNet):
+    def __init__(self):
+        super().__init__()
+        self.conv1 = nn.Conv2d(3, 3, kernel_size=1, stride=1)
+        self.conv2 = nn.Conv2d(3, 3, kernel_size=1, stride=1)
+
+    def forward(self, x):
         x = self.conv1(x)
-        x = self.identity(x)
         x = self.conv2(x)
-        x = self.identity(x)
+        x = self.conv1(x)
         x = self.conv2(x)
-        x = self.identity(x)
         x = self.conv2(x)
-        x = self.identity(x)
-        y = self.conv2(y)
-        y = self.identity(y)
-        y = self.conv2(y)
-        y = self.identity(y)
-        y = self.conv1(y)
-        y = self.identity(y)
-        y = self.conv1(y)
-        y = self.identity(y)
-        y = self.conv1(y)
-        return x - y, y - x
-
-
-class ReuseLayerNetTest(BasePytorchTest):
-    """
-    This test checks:
-    The reuse of a layer in a model.
-    """
+        return x
+
+class ReuseFunctionalLayerNet(BaseReuseLayerNet):
+    def __init__(self):
+        super().__init__()
+        self.conv1_weight = nn.Parameter(torch.randn(3, 3, 2, 2))
+        self.conv1_bias = nn.Parameter(torch.randn(3))
+        self.conv2_weight = nn.Parameter(torch.randn(3, 3, 2, 2))
+        self.conv2_bias = nn.Parameter(torch.randn(3))
+
+    def forward(self, x):
+        x = F.conv2d(x, self.conv1_weight, self.conv1_bias, stride=1, groups=1)
+        x = F.conv2d(x, self.conv1_weight, self.conv1_bias, stride=1, groups=1)
+        x = F.conv2d(x, self.conv2_weight, self.conv2_bias, stride=1, groups=1)
+        x = F.conv2d(x, self.conv2_weight, self.conv2_bias, stride=1, groups=1)
```

Comment:
I would also check that identical weight but different bias and vice versa are not considered reused